### PR TITLE
priming tool

### PR DESCRIPTION
**Pull request by evgensheff**
## Description

Tool for performing priming without further block sync

## Type of change

- [ ] New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: cmd/util-primer/db/run_primer.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+	}
+
+	extensionList = append(extensionList, []executor.Extension[txcontext.TxContext]{
+		register.MakeRegisterProgress(cfg, 100_000),
+		// RegisterProgress should be the as top-most as possible on the list
+		// In this case, after StateDb is created.
+		// Any error that happen in extension above it will not be correctly recorded.
+		logger.MakeProgressLogger[txcontext.TxContext](cfg, 15*time.Second),
+		logger.MakeErrorLogger[txcontext.TxContext](cfg),
+		tracker.MakeBlockProgressTracker(cfg, 100_000),
```
Comment (wsodsong) [inline cmd/util-primer/db/run_primer.go:71]: These logger extensions log events when processing blocks or transactions (Pre/PostBlock and Pre/PostTransaction). This command does not trigger PreBlock or PreTransaction, I don't think anything will be logged and can be removed.

File: executor/executor.go [hunk 2]
```diff
@@ -536,6 +536,14 @@ func (e *executor[T]) runBlocks(params Params, processor Processor[T], extension
 	return err
 }
 
+func PreRun[T any](params Params, extensions []Extension[T]) (err error) {
+	state := State[T]{}
+	ctx := Context{State: params.State}
+
+	state.Block = params.From
+	return signalPreRun(state, &ctx, extensions)
```
Comment (wsodsong) [inline executor/executor.go:544]: For completeness, I recommend supporting both opening and closing of a scope -- i.e., `signalPreRun` and `signalPostRun`.

File: cmd/util-primer/db/run_primer.go [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+}
+
+func runPriming(
+	cfg *utils.Config,
+) error {
+	var extensionList = []executor.Extension[txcontext.TxContext]{
+		profiler.MakeCpuProfiler[txcontext.TxContext](cfg),
+		profiler.MakeDiagnosticServer[txcontext.TxContext](cfg),
+		statedb.MakeStateDbManager[txcontext.TxContext](cfg, ""),
+		statedb.MakeLiveDbBlockChecker[txcontext.TxContext](cfg),
```
[the remark this answers is not in the document]
Comment (evgensheff) [reply cmd/util-primer/db/run_primer.go]: @matejmlejnek MakeStateDbManager initialises stateDb, without it the program panics

File: cmd/util-primer/db/run_primer.go (hunk 1 above)
Comment (evgensheff) [reply cmd/util-primer/db/run_primer.go:73]: Got it

File: executor/executor.go (hunk 2 above)
Comment (evgensheff) [reply executor/executor.go]: Thank you, that makes sense

File: executor/executor.go [hunk 4]
```diff
@@ -536,6 +536,21 @@ func (e *executor[T]) runBlocks(params Params, processor Processor[T], extension
 	return err
 }
 
+func PreRun[T any](params Params, extensions []Extension[T]) (err error) {
+	state := State[T]{}
+	ctx := Context{State: params.State}
+
+	state.Block = params.To
```
[the remark this answers is not in the document]
Comment (evgensheff) [reply executor/executor.go:543]: Not for all, func PreRun only used in priming subcommand

File: Makefile [hunk 5]
```diff
@@ -98,6 +98,7 @@ util-updateset: carmen tosca
 
 util-db: carmen tosca
 	GOPROXY=$(GOPROXY) \
+	CGO_CFLAGS="-g -O2  -DMDBX_FORCE_ASSERTIONS=1 -Wno-error=strict-prototypes" \
```
[the remark this answers is not in the document]
Comment (evgensheff) [reply Makefile:101]: It didn't work on my mac without it. At your discretion

File: executor/executor.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (wsodsong) [reply executor/executor.go:543]: I feel that the name `PreRun` is misleading. This is a new function runs by `primer`. The line doesn't affect other commands. I would rename the function to `runUtilPrimer` or something more specific.

[the remark this answers is not in the document]
Comment (evgensheff) [reply executor/executor.go:543]: That makes sense, thanks

**Review by wsodsong — APPROVED** [review]

**Pull request merged**